src/OrleansServiceBus/Providers/Streams/EventHub/EventHubDataAdapter.cs

```diff
@@ -267,36 +270,6 @@ public virtual StreamPosition GetStreamPosition(EventData queueMessage)
             return new StreamPosition(stremIdentity, token);
         }
 
-        /// <summary>
-        /// Given a purge request, indicates if a cached message should be purged from the cache
-        /// </summary>
-        /// <param name="cachedMessage"></param>
-        /// <param name="newestCachedMessage"></param>
-        /// <param name="purgeRequest"></param>
-        /// <param name="nowUtc"></param>
-        /// <returns></returns>
-        public bool ShouldPurge(ref CachedEventHubMessage cachedMessage, ref CachedEventHubMessage newestCachedMessage, IDisposable purgeRequest, DateTime nowUtc)
-        {
-            // if we're purging our current buffer, don't use it any more
-            var purgedResource = (FixedSizeBuffer)purgeRequest;
-            if (currentBuffer != null && currentBuffer.Id == purgedResource.Id)
-            {
-                currentBuffer = null;
-            }
-
-            TimeSpan timeInCache = nowUtc - cachedMessage.DequeueTimeUtc;
-            // age of message relative to the most recent event in the cache.
-            TimeSpan relativeAge = newestCachedMessage.EnqueueTimeUtc - cachedMessage.EnqueueTimeUtc;
-
-            return ShouldPurgeFromResource(ref cachedMessage, purgedResource) || timePurage.ShouldPurgFromTime(timeInCache, relativeAge);
-        }
-
-        private static bool ShouldPurgeFromResource(ref CachedEventHubMessage cachedMessage, FixedSizeBuffer purgedResource)
-        {
-            // if message is from this resource, purge
-            return cachedMessage.Segment.Array == purgedResource.Id;
-        }
-
```

Comment:
it's been removed. this is the part which caused the low traffic queue bug. 
this code basically tell: no matter whether this message is old enough to be purged by TimeBasedPurgePredicate or not, purge it as long as it belongs to this block.  While the last block in block pool doesn't mean all messages on it should be time based purge.